File: src/main/webapp/tutorials/code/java/arrays-basics.java
public class Main {
    public static void main(String[] args) {
        // 1. Declaration & Instantiation
        // Creates an array of 5 integers, initialized to 0
        int[] numbers = new int[5];

        // 2. Assigning values
        numbers[0] = 10;
        numbers[1] = 20;
        numbers[2] = 30;
        numbers[3] = 40;
        numbers[4] = 50;

        System.out.println("First element: " + numbers[0]);
        System.out.println("Last element: " + numbers[4]);

        // 3. Array Length Property
        System.out.println("Array Size: " + numbers.length);

        // 4. Looping through array
        System.out.print("All elements: ");
        for (int i = 0; i < numbers.length; i++) {
            System.out.print(numbers[i] + " ");
        }
        System.out.println();

        // 5. Index Out of Bounds Error
        // numbers[5] = 60; // Throws ArrayIndexOutOfBoundsException!
    }
}

File: src/main/webapp/tutorials/code/php/arrays-basics.php
<?php
// PHP Array Basics Demo
// 8gwifi.org/tutorials

echo "=== Creating Indexed Arrays ===\n";

// Short syntax (PHP 5.4+) - Recommended
$fruits = ["Apple", "Banana", "Cherry", "Date"];

// Traditional syntax
$colors = array("Red", "Green", "Blue");

echo "Fruits: ";
print_r($fruits);

echo "\n=== Accessing Array Elements ===\n";
echo "First fruit: " . $fruits[0] . "\n";
echo "Second fruit: " . $fruits[1] . "\n";
echo "Last fruit: " . $fruits[count($fruits) - 1] . "\n";

// Negative index (PHP 8.0+ allows with array functions)
echo "Last using end(): " . end($fruits) . "\n";

echo "\n=== Modifying Arrays ===\n";
$fruits[1] = "Blueberry";  // Replace element
echo "After changing index 1: ";
print_r($fruits);

$fruits[] = "Elderberry";  // Append element
echo "After appending: ";
print_r($fruits);

echo "\n=== Array Length ===\n";
echo "Number of fruits: " . count($fruits) . "\n";
echo "Using sizeof(): " . sizeof($fruits) . "\n";

echo "\n=== Checking if Array ===\n";
$notArray = "Hello";
echo "Is \$fruits an array? " . (is_array($fruits) ? "Yes" : "No") . "\n";
echo "Is \$notArray an array? " . (is_array($notArray) ? "Yes" : "No") . "\n";

echo "\n=== Empty Arrays ===\n";
$empty1 = [];
$empty2 = array();
echo "Empty array count: " . count($empty1) . "\n";
echo "Is empty? " . (empty($empty1) ? "Yes" : "No") . "\n";

echo "\n=== Quick Array Creation ===\n";
$numbers = range(1, 5);
echo "range(1, 5): ";
print_r($numbers);

$letters = range('a', 'e');
echo "range('a', 'e'): ";
print_r($letters);

$evens = range(2, 10, 2);
echo "range(2, 10, 2): ";
print_r($evens);
?>
